### PR TITLE
Fix: Translation for system message

### PR DESCRIPTION
**Pull request by tienifr**
### Details

System messages are currently hard-coded strings. This PR added `translationKey` for them.

### Fixed Issues

$ https://github.com/Expensify/App/issues/19449
PROPOSAL: https://github.com/Expensify/App/issues/19449#issuecomment-1561043031

### Tests

1. Login with any account
2. Go to Settings > Preferences > Language
3. Select Spanish language and go back
4. Go to any chat
5. Send an Attachment
6. Verify that `[Archivo adjunto]` shows on LHN

- [x] Verify that no errors appear in the JS console

### Offline tests

NA

### QA Steps

1. Login with any account
2. Go to Settings > Preferences > Language
3. Select Spanish language and go back
4. Go to any chat
5. Send an Attachment
6. Verify that `[Archivo adjunto]` shows on LHN

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

[Screencast from 28-06-2023 16:53:59.webm](https://github.com/Expensify/App/assets/113963320/504faed2-9b70-46ee-996d-c407cd50bd05)

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/113963320/3281aa01-fae4-4d57-b500-ca4faac4b0d0

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/113963320/eb97ae49-97fb-4f86-8496-fcf14c22d1a4

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/113963320/013e50f5-dadb-4936-a4be-a684771f7b37

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/113963320/8aacde93-9a93-4fa8-9d7e-40a81054aa27

</details>

<details>
<summary>Android</summary>

[Screencast from 07-07-2023 23:46:56.webm](https://github.com/Expensify/App/assets/113963320/7b17d04d-63a6-450b-a594-c930bcf8c1ba)

</details>

### CODE REVIEW

**Comment by yuwenmemon** [comment]
@tienifr I think we can proceed without worrying about the old messages if that makes things simpler.

**Comment by yuwenmemon** [comment]
Ah, okay then I guess so, but then why would you need this?

> Add translationKey field under message to all old report actions. Defaults to empty string, if attachment then common.attachment.

**Comment by abdulrahuman5196** [comment]
Just providing some information on the subtitle translation issues discussion we had previously for LHN since I saw this PR which was mentioned in another issue. @yuwenmemon @eVoloshchak 
We are having different translation issues and previously discussed here - https://expensify.slack.com/archives/C01GTK53T8Q/p1687190693298329


The last callout was to avoid complicated solutions incase of LHN translations and leave out small bugs if only way forward is a complex solution. Even attachments was mentioned there and the current path seems to have backend changes and complex solutions.
And one more concern is that, it not just we have either translated or non-translated subtitles, we also have complex subtitles where the message has part translated and part non-translated content, which would not have solved by the new translation key and we can't use the translation key in this case. For Eg.
https://github.com/Expensify/App/blob/59973575da78597ccac89781b4d7117110b82078/src/libs/SidebarUtils.js#L342-L358

**But anyways just providing my 2 cents from the last discussion I had and leaving it up for you guys to decide.**

**Comment by bernhardoj** [comment]
The proposal 
here https://github.com/Expensify/App/issues/19449#issuecomment-1559820634 is actually a straight and easy solution.

**Comment by eVoloshchak** [comment]
@abdulrahuman5196, thanks for attaching the discussion!
According to the thread: 
>  would be totally fine if server side messages are not translated ever, since translating the backend is not something we did yet and whenever we want to fully internationalize the app, we will need to add backend translation anyway
I am also fine leaving small bugs related to this, especially when switching languages

It is a rare bug that can be reproduced only when switching languages, I agree this isn't a top priority. I think the implementation in this PR is correct, so in case we decide to implement this later - we should use it. 
Asking if we should proceed in [slack thread](https://expensify.slack.com/archives/C01GTK53T8Q/p1688134958591739?thread_ts=1687190693.298329&cid=C01GTK53T8Q)

**Comment by eVoloshchak** [comment]
@tienifr, the translation in LHN works well now.
I think we should translate "Uploading attachment" message too

https://github.com/Expensify/App/assets/9059945/9ceaa359-d944-4b5b-8ae6-45d6978dae99

**Review by eVoloshchak — CHANGES_REQUESTED** [review]
@tienifr, the translation in LHN works well now.
I think we should translate "Uploading attachment" message too

https://github.com/Expensify/App/assets/9059945/9ceaa359-d944-4b5b-8ae6-45d6978dae99

**Comment by eVoloshchak** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by eVoloshchak** [comment]
> We are going to remove the Uploading attachment message, aren't we? I thought we have already agreed on translating [Attachment] only.

Ah, you're correct, we indeed have an issue for that

Looks good to me, there are only two steps left
1. Merge with main
2. Upload videos for the remaining platforms

**Review by eVoloshchak — APPROVED** [review]
LGTM

**Review by yuwenmemon — APPROVED** [review]

File: src/libs/actions/Report.js [hunk 1]
```diff
@@ -249,6 +249,7 @@ function addActions(reportID, text = '', file) {
 
     const optimisticReport = {
         lastVisibleActionCreated: currentTime,
+        lastMessageTranslationKey: lodashGet(lastAction, 'message[0].translationKey', ''),
```
Comment (eh2077) [inline src/libs/actions/Report.js:252]: It looks like that push event should also update this field to update let LHN message after sending an attachment, like this case https://github.com/Expensify/App/issues/22937

File: src/libs/isReportMessageAttachment.js [hunk 2]
```diff
@@ -4,10 +4,14 @@ import CONST from '../CONST';
  * Check whether a report action is Attachment or not.
  * Ignore messages containing [Attachment] as the main content. Attachments are actions with only text as [Attachment].
  *
- * @param {Object} reportActionMessage report action's message as text and html
+ * @param {Object} reportActionMessage report action's message as text, html and translationKey
  * @returns {Boolean}
  */
-export default function isReportMessageAttachment({text, html}) {
+export default function isReportMessageAttachment({text, html, translationKey}) {
+    if (translationKey) {
```
Comment (eVoloshchak) [inline src/libs/isReportMessageAttachment.js:11]: 👋 This was partially responsible for a regression in https://github.com/Expensify/App/issues/24246
It was mainly a back-end problem, the translation key is not set/reset when we change from a message _with_ a translation key to one _without_, or vice-versa
While this wasn't caused by this PR, leaving a note to myself and others to test more complex test cases (this one included testing with two accounts)

**Pull request merged**